imagenet/main.py

```diff
@@ -415,7 +415,7 @@ def update(self, val, n=1):
         self.avg = self.sum / self.count
 
     def all_reduce(self):
-        total = torch.FloatTensor([self.sum, self.count])
+        total = torch.Tensor([self.sum, self.count], device=self.sum.device, dtype=torch.float32)
```

Comment:
Let's use `torch.tensor`. I'm curious where self.sum is set to be a tensor(i.e. so we can use device attribute). 

```suggestion
        total = torch.tensor([self.sum, self.count], dtype=torch.float32, device=self.sum.device)
```